synapse/storage/databases/main/roommember.py

```diff
@@ -886,6 +886,40 @@ def _is_local_host_in_room_ignoring_users_txn(txn):
             _is_local_host_in_room_ignoring_users_txn,
         )
 
+    def get_stream_ordering_for_local_membership_txn(
+        self,
+        txn: LoggingTransaction,
+        user_id: str,
+        room_id: str,
+        membership: Membership,
+    ) -> Optional[int]:
+        """Get the stream ordering for a given local room membership.
+
+        Args:
+            user_id: The user ID to retrieve the membership for.
+            room_id: The room ID to retrieve the membership for.
+            membership: The membership to retrieve the stream ordering for.
+
+        Returns:
+            The stream ordering, or None if the membership wasn't found.
+        """
+        txn.execute(
+            """
+            SELECT stream_ordering FROM local_current_membership
+            LEFT JOIN events USING (event_id, room_id)
+            WHERE membership = ?
+                AND user_id = ?
+                AND room_id = ?
+            """,
+            (membership, user_id, room_id),
+        )
+        row = txn.fetchone()
+
+        if row is None:
+            return None
+
+        return row[0]
```

Comment:
this doesn't really make much sense, conceptually. How can a "room membership" have a stream ordering? `local_current_membership` is meant to be looked up by `user_id` and `room_id`, not membership: what is the effect of passing in a `membership` praram that does not correspond to the user's current membership?

Reply:
> this doesn't really make much sense, conceptually. How can a "room membership" have a stream ordering?

Agreed that the name is meh. What it does is "get the stream ordering associated with the event for the latest membership update from this user in this room" but I'm bad at translating this kind of things into function names.

I'll probably rework this function though, possibly making it work with the new function from https://github.com/matrix-org/synapse/pull/8112.

Reply:
> `local_current_membership` is meant to be looked up by `user_id` and `room_id`, not membership

Ah right, `is_local_host_in_room_ignoring_users` is requesting via membership so I think that's why I've been using it.